pdpxc/pdpxc-pxco-integration-scheduler.groovy

```diff
@@ -0,0 +1,106 @@
+                    withCredentials([[$class: 'AmazonWebServicesCredentialsBinding', accessKeyVariable: 'AWS_ACCESS_KEY_ID', credentialsId: 'AMI/OVF', secretKeyVariable: 'AWS_SECRET_ACCESS_KEY']]) {
+                        sh """
+                            aws s3 cp ${S3_BUCKET}/${TRIGGER_FILE} ${LOCAL_TRIGGER_FILE}
+                        """
+                    
+                    }
+                    
+                }
+            }
+        }
+
+        stage("Check"){
+            steps{
+                script{
+                    def pxcoValue = sh(script: "grep '^PXCO=' ${LOCAL_TRIGGER_FILE} | cut -d '=' -f2", returnStdout: true).trim()
+                    
+                    if (pxcoValue) {
+                        echo "PXCO value: ${pxcoValue}"
+                        
+                        if (pxcoValue == "0") {
+                            echo "PXCO is set to 0. Setting build status to UNSTABLE."
+                            currentBuild.result = 'UNSTABLE'
+                        } else if (pxcoValue == "1") {
+
+                            def jobId = env.BUILD_ID
+                            
+                            // Modify and upload the file back to S3
+                            
+                            withCredentials([[$class: 'AmazonWebServicesCredentialsBinding', accessKeyVariable: 'AWS_ACCESS_KEY_ID', credentialsId: 'AMI/OVF', secretKeyVariable: 'AWS_SECRET_ACCESS_KEY']]) {
+                            
+                                sh """
+                                    sed -i 's/^PXCO=.*/PXCO=0/' ${LOCAL_TRIGGER_FILE}
+                                    aws s3 cp ${LOCAL_TRIGGER_FILE} ${S3_BUCKET}/${TRIGGER_FILE}
+                                """
+                            }
+                            
+                            //currentBuild.result = 'SUCCESS'  // Explicitly set build status to SUCCESS
+                        
+
+                            // Trigger the PXC Operator Jobs on cloud.cd jenkins
+                            def envFile = readFile("${LOCAL_TRIGGER_FILE}").trim()
+                            envFile.readLines().each { line ->
+                                def parts = line.split('=', 2)
+                                if (parts.size() == 2) {
+                                    def key = parts[0].trim()
+                                    def value = parts[1].trim()
+                                    env."${key}" = value  // Workaround to avoid putAt()
+                                }
+                            }
+                            echo "Environment variables set!"
+                            
+                            build(
+                                job: 'pxco-gke-1',
+                                parameters: [
+                                    string(name: "TEST_SUITE", value: "run-distro.csv"),
+                                    string(name: "IGNORE_PREVIOUS_RUN", value: "YES"),
```

Comment:
Please pass NO here. This will be the first run with those images and branch so all the test should run and rerunning failed ones will be easier.
Can we implement retry if the 1st run fails?